de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/handlers/NewFolderHandler.java

```diff
@@ -1,50 +1,37 @@
- 
+
 package de.tu_bs.cs.isf.e4cf.parts.project_explorer.handlers;
 
-import java.nio.file.Path;
+import java.io.File;
 
+import org.eclipse.e4.core.contexts.IEclipseContext;
 import org.eclipse.e4.core.di.annotations.CanExecute;
 import org.eclipse.e4.core.di.annotations.Execute;
-import org.eclipse.swt.graphics.Point;
 import org.eclipse.swt.widgets.Shell;
 
 import de.tu_bs.cs.isf.e4cf.core.file_structure.FileTreeElement;
-import de.tu_bs.cs.isf.e4cf.core.file_structure.WorkspaceFileSystem;
-import de.tu_bs.cs.isf.e4cf.core.file_structure.util.FileHandlingUtility;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPDialogService;
-import de.tu_bs.cs.isf.e4cf.core.util.services.RCPImageService;
+import de.tu_bs.cs.isf.e4cf.core.util.ServiceContainer;
 import de.tu_bs.cs.isf.e4cf.core.util.services.RCPSelectionService;
-import de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.NewFolderPage;
-import de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.NewFolderWizard;
 
-public class NewFolderHandler {
-	
+public class NewFolderHandler implements IHandler {
+
+	private static final String FOLDER_PLACEHOLDER = "/New Folder";
+
 	@Execute
-	public void execute(Shell shell, RCPDialogService dialogService, RCPSelectionService selectionService, 
-			RCPImageService imageService, WorkspaceFileSystem fileSystem) {
-		FileTreeElement selectedElement = selectionService.getCurrentSelectionFromExplorer();
-		
-		NewFolderPage page = new NewFolderPage(
-				"NewFolder", "Choose a Folder Name", 
-				imageService.getImageDescriptor(null, "icons/Explorer_View/items/folder24.png")
-		);
-		
-		// create selected directory when selected element available, else create new project
-		Path directory;
-		if (selectedElement != null) {
-			directory = FileHandlingUtility.getPath(selectedElement);
-		} else {
-			directory = FileHandlingUtility.getPath(fileSystem.getWorkspaceDirectory());
+	public void execute(IEclipseContext context, ServiceContainer services, Shell shell) {
+		FileTreeElement element = services.rcpSelectionService.getCurrentSelectionFromExplorer();
+		String stringPath = element.getAbsolutePath();
+		File folderToCreate = new File(stringPath + FOLDER_PLACEHOLDER);
```

Comment:
use `Paths.get` Method to concat multiple paths.

Reply:
both parts are string and new file also asks for a string, so Paths would  be too round-about. Should i change it so that both are paths to begin with?

Reply:
`Paths.get` accepts multiple strings from which you want to build up the path and with `getabsolutpath` you can get a string representation. I would suggest to use this because it automatically uses the correct vendor path seperator

Reply:
good point with the vendor, changed it to Paths.get()